### PR TITLE
Enable seamless playback of multipart Movies.

### LINKED ISSUE #4209: Movies with Additional Parts don't automatically play next file.
**Describe The Bug**

Dupe / recreation of https://github.com/jellyfin/jellyfin/issues/8090 in jellyfin-web project.

Feature request:

When a movie has multiple parts, each part must be explicitly played rather than playing the movie automatically playing the next part when playback of the nth completes.

Ideally playing the movie will allow the user to play each part one after the other without explicitly playing part1, then returning to the details page and playing part 2, and so on.

**System (please complete the following information):**
 - Platform: [Linux]
 - Browser: [Chrome]
 - Jellyfin Version: [e.g. 10.8.5]

**Additional Context**

### PR DESCRIPTION
**Pull request by farraherbg**
**Changes**

    Adds support for seamless playback of multipart Movies via
    Add getAdditionalParts method to playbackmanager
    When a movie with an additional part is played,
    enqueue the additional parts in the playlist.

**Issues**

Fixes #4209

### CODE REVIEW

File: src/components/playback/playbackmanager.js [hunk 1]
```diff
@@ -1888,7 +1888,10 @@ class PlaybackManager {
 
             if (options.items) {
                 return translateItemsForPlayback(options.items, options).then(function (items) {
-                    return playWithIntros(items, options);
+                    getAdditionalParts(items).then(function (allItems) {
+                        const flattened = allItems.flatMap(i => i);
+                        return playWithIntros(flattened, options);
+                    });
```
Comment (dmitrylyzo) [inline src/components/playback/playbackmanager.js:1894]: Not tested, but there will be less nested Promises this way.
```
return translateItemsForPlayback(options.items, options)
    .then(getAdditionalParts)
    .then(function (allItems) {
        const flattened = allItems.flatMap(i => i);
        return playWithIntros(flattened, options);
    });
```

File: src/components/playback/playbackmanager.js [hunk 2]
```diff
@@ -2039,6 +2045,20 @@ class PlaybackManager {
             return player.play(options);
         }
 
+        const getAdditionalParts = async (items) => {
+            const getOneAdditionalPart = async function (item) {
+                if (item.Type === 'Movie') {
+                    const client = ServerConnections.getApiClient(item.ServerId);
+                    const user = await client.getCurrentUser();
+                    const additionalParts = await client.getAdditionalVideoParts(user.Id, item.Id);
+                    return (additionalParts.Items.length) ? [item, ...additionalParts.Items] : [item];
```
Comment (dmitrylyzo) [inline src/components/playback/playbackmanager.js:2054]: Does this work?
```suggestion
                    if (additionalParts.Items.length) return [item, ...additionalParts.Items];
```
^ one exit for simple items.

File: src/components/playback/playbackmanager.js (hunk 1 above)
Comment (farraherbg) [reply src/components/playback/playbackmanager.js]: Done

**Review by dmitrylyzo — COMMENTED** [review]
PR seems to work.
It would be better to rebase and squash the commits.

**Review by dmitrylyzo — APPROVED** [review]
I rebased and squashed the commits. You need to update the reference in your working copy if PR requires additional changes.

**Pull request merged**